test/auto-take-profit.ts

```diff
@@ -0,0 +1,564 @@
+            collateralAmount = collateral.toString()
+            debtAmount = debt.toString()
+            nextCollRatioAsPercentage = Math.floor(parseFloat(collRatio) * 100)
+
+            nextPrice = await McdViewInstance.getNextPrice(ethAIlk)
+
+            cdpData = {
+                gemJoin: hardhatUtils.addresses.MCD_JOIN_ETH_A,
+                fundsReceiver: receiverAddress,
+                cdpId: testCdpId,
+                ilk: '0x0000000000000000000000000000000000000000000000000000000000000000',
+                requiredDebt: 0, // can stay 0 overriden in SC anyway
+                borrowCollateral: collateralAmount,
+                withdrawCollateral: 0,
+                withdrawDai: 0,
+                depositDai: 0, // simple case no additional dai
+                depositCollateral: 0,
+                skipFL: false,
+                methodName: '',
+            }
+
+            exchangeData = {
+                fromTokenAddress: hardhatUtils.addresses.WETH,
+                toTokenAddress: hardhatUtils.addresses.DAI,
+                fromTokenAmount: '',
+                toTokenAmount: '',
+                minToTokenAmount: '',
+                exchangeAddress: '',
+                _exchangeCalldata: '',
+            }
+        })
+
+        describe('closeToCollateral operation', async () => {
+            before(async () => {
+                const debt = EthersBN.from(debtAmount)
+                const tradeSize = debt.mul(10020).div(10000) // + our fee 0.2%
+                // const collateralValue = debt.mul(nextCollRatioAsPercentage).div(100)
+
+                exchangeData.fromTokenAmount = collateralAmount
+                exchangeData.minToTokenAmount = tradeSize.toString()
+                exchangeData.toTokenAmount = EthersBN.from(exchangeData.minToTokenAmount).mul(102).div(100).toString() // slippage 2%
+                exchangeData.exchangeAddress = ONE_INCH_V4_ROUTER
+                exchangeData._exchangeCalldata = forgeUnoswapCalldata(
+                    hardhatUtils.addresses.WETH,
+                    exchangeData.fromTokenAmount,
+                    exchangeData.minToTokenAmount,
+                )
+            })
+
+            describe('when Trigger is above next col ratio', async () => {
+                let triggerId: number
+                let triggerData: BytesLike
+                let executionData: BytesLike
+                let signer: Signer
+
+                before(async () => {
+                    // makeSnapshot
+                    snapshotId = await hre.ethers.provider.send('evm_snapshot', [])
+                    signer = await hardhatUtils.impersonate(proxyOwnerAddress)
+                    // addTrigger
+                    triggerData = encodeTriggerData(
+                        testCdpId,
+                        TriggerType.AUTO_TP_COLLATERAL,
+                        nextPrice.add('32665974900000000000'),
```

Comment:
what is this magic number ?

Reply:
ah I've tried targeting price above next price. Now irrelevant. Changed to 1000